rocketpy/simulation/monte_carlo.py

```diff
@@ -159,22 +197,360 @@ def simulate(self, number_of_simulations, append=False):
+            input_file = open(self._input_file, open_mode, encoding="utf-8")
+            output_file = open(self._output_file, open_mode, encoding="utf-8")
+            error_file = open(self._error_file, open_mode, encoding="utf-8")
+        else:
+            input_file = h5py.File(Path(self._input_file).with_suffix(".h5"), open_mode)
+            output_file = h5py.File(
+                Path(self._output_file).with_suffix(".h5"), open_mode
+            )
+            error_file = open(self._error_file, open_mode, encoding="utf-8")
+
+            idx_i = self.__get_initial_sim_idx(input_file, light_mode=light_mode)
+            idx_o = self.__get_initial_sim_idx(output_file, light_mode=light_mode)
+
+        if idx_i != idx_o:
+            raise ValueError(
+                "Input and output files are not synchronized. Append mode is not available."
+            )
+
+        # Run simulations
         try:
             while self.iteration_count < self.number_of_simulations:
-                self.__run_single_simulation(input_file, output_file)
+                self.__run_single_simulation(
+                    self.iteration_count + idx_i,
+                    input_file,
+                    output_file,
+                    light_mode=light_mode,
+                )
+
         except KeyboardInterrupt:
             print("Keyboard Interrupt, files saved.")
             error_file.write(json.dumps(self._inputs_dict, cls=RocketPyEncoder) + "\n")
             self.__close_files(input_file, output_file, error_file)
+
         except Exception as error:
             print(f"Error on iteration {self.iteration_count}: {error}")
             error_file.write(json.dumps(self._inputs_dict, cls=RocketPyEncoder) + "\n")
             self.__close_files(input_file, output_file, error_file)
             raise error
 
-        self.__finalize_simulation(input_file, output_file, error_file)
+        self.__finalize_simulation(
+            input_file, output_file, error_file, light_mode=light_mode
+        )
+
+    def __run_in_parallel(self, append, light_mode, n_workers=None):
+        """
+        Runs the monte carlo simulation in parallel.
+
+        Parameters
+        ----------
+            append: bool
+                If True, the results will be appended to the existing files. If
+                False, the files will be overwritten.
+            light_mode: bool
+                If True, only variables from the export_list will be saved to
+                the output file as a .txt file. If False, all variables will be
+                saved to the output file as a .h5 file.
+            n_workers: int, optional
+                Number of workers to be used. If None, the number of workers
+                will be equal to the number of CPUs available. Default is None.
+
+        Returns
+        -------
+        None
+        """
+        processes = []
+
+        if n_workers is None:
+            n_workers = os.cpu_count()
```

Comment:
I think this approach may be unsafe, as it may cause the program stop responding to the user. I would suggest for us to take 75% of the number of threads instead of 100%.

```suggestion
            n_workers = int((3 / 4) * os.cpu_count())
```

Reply:
Indeed it's a good suggestion to spare one or two workers when the user doesn't provide the number of workers, but I'd go for a fixed amount of cpus, otherwise we could undermine the performance too much.